lib/json_schemer/schema/base.rb

```diff
@@ -615,13 +626,19 @@ def resolve_ids(schema, ids = {}, parent_uri = nil, pointer = '')
         if schema.is_a?(Array)
           schema.each_with_index { |subschema, index| resolve_ids(subschema, ids, parent_uri, "#{pointer}/#{index}") }
         elsif schema.is_a?(Hash)
-          id = schema[id_keyword]
-          uri = join_uri(parent_uri, id)
-          unless uri == parent_uri
-            ids[uri.to_s] = {
-              schema: schema,
-              pointer: pointer
-            }
+          uri = join_uri(parent_uri, schema[id_keyword])
+          schema.each_pair do |key, value| 
```

Comment:
Not sure what `each_pair` is, but plain `each` is used for this below.

Reply:
it's the same thing, it passes key, value to the block. I think its just one of those weird redundant methods that somehow I got used to using, heh I'll update it